src/action-creators/undoArchive.js

```diff
@@ -22,6 +35,14 @@ export const undoArchive = ({ originalPath, currPath, offset }) => dispatch => {
     offset
   })
 
+  if (getThoughtsRanked(archiveContext).length === 0) {
+    dispatch({
+      type: 'existingThoughtDelete',
+      context: contextOfArchive,
+      thoughtRanked: head(rankThoughtsFirstMatch(archiveContext))
```

Comment:
Rather than recreating the ranks from the context, which is expensive and does not work if there are duplicate thoughts, build the correct thought using `currPath`, `contextOf`, and `getThoughtsRanked`. 

`rankThoughtsFirstMatch` should be avoided if at all possible. It's mainly used for guessing the cursor from the url, although it has admittedly leaked out into other code.

Reply:
Removed `rankThoughtsFirstMatch`.